c_glib/parquet-glib/arrow-file-writer.cpp

```diff
@@ -32,10 +32,274 @@ G_BEGIN_DECLS
+
+  G_OBJECT_CLASS(gparquet_writer_properties_parent_class)->finalize(object);
+}
+
+static void
+gparquet_writer_properties_init(GParquetWriterProperties *object)
+{
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(object);
+  priv->builder = new parquet::WriterProperties::Builder();
+  priv->changed = TRUE;
+}
+
+static void
+gparquet_writer_properties_class_init(GParquetWriterPropertiesClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->finalize     = gparquet_writer_properties_finalize;
+}
+
+/**
+ * gparquet_writer_properties_new:
+ *
+ * Return: A newly created #GParquetWriterProperties.
+ *
+ * Since: 1.0.0
+ */
+GParquetWriterProperties *
+gparquet_writer_properties_new(void)
+{
+  auto writer_properties = g_object_new(GPARQUET_TYPE_WRITER_PROPERTIES,
+                                        NULL);
+  return GPARQUET_WRITER_PROPERTIES(writer_properties);
+}
+
+/**
+ * gparquet_writer_properties_set_compression:
+ * @properties: A #GParquetWriterProperties.
+ * @compression_type: A #GArrowCompressionType.
+ *
+ * Since: 1.0.0
+ */
+void
+gparquet_writer_properties_set_compression(GParquetWriterProperties *properties,
+                                           GArrowCompressionType compression_type)
+{
+  auto arrow_compression_type = garrow_compression_type_to_raw(compression_type);
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(properties);
+  priv->builder->compression(arrow_compression_type);
+  priv->changed = TRUE;
+}
+
+/**
+ * gparquet_writer_properties_get_compression:
+ * @properties: A #GParquetWriterProperties.
+ * @dotstring: The dot string path.
+ *
+ * Returns: The compression type of #GParquetWriterProperties.
+ *
+ * Since: 1.0.0
+ */
+GArrowCompressionType
+gparquet_writer_properties_get_compression(GParquetWriterProperties *properties,
+                                           gchar *dotstring)
+{
+  auto parquet_properties = gparquet_writer_properties_get_raw(properties);
+  auto parquet_column_path = parquet::schema::ColumnPath::FromDotString(dotstring);
+  auto arrow_compression = parquet_properties->compression(parquet_column_path);
+  return garrow_compression_type_from_raw(arrow_compression);
+}
+
+/**
+ * gparquet_writer_properties_enable_dictionary:
+ * @properties: A #GParquetWriterProperties.
+ *
+ * Since: 1.0.0
+ */
+void
+gparquet_writer_properties_enable_dictionary(GParquetWriterProperties *properties)
+{
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(properties);
+  priv->builder->enable_dictionary();
+  priv->changed = TRUE;
+}
+
+/**
+ * gparquet_writer_properties_disable_dictionary:
+ * @properties: A #GParquetWriterProperties.
+ *
+ * Since: 1.0.0
+ */
+void
+gparquet_writer_properties_disable_dictionary(GParquetWriterProperties *properties)
+{
+  auto priv = GPARQUET_WRITER_PROPERTIES_GET_PRIVATE(properties);
+  priv->builder->disable_dictionary();
+  priv->changed = TRUE;
+}
+
+/**
+ * gparquet_writer_properties_dictionary_enabled:
```

Comment:
Could you add `is_` for predicate?
`gparquet_writer_properties_is_dictionary_enabled`.